packages/ui-components/src/lib/hooks/usePersistedFormValues.ts

```diff
@@ -0,0 +1,15 @@
+import { useEffect } from 'react';
+import _isEqual from 'lodash/isEqual';
+import _debounce from 'lodash/debounce';
+import { usePrevious } from './usePrevious';
+
+export const usePersistedFormValues = <T>(values: T, persistFunction?: (values: T) => void, debounce?: number) => {
```

Comment:
Wondering if we should have a default `persistFunction`, if they don't pass one in -- to support these we would need to allow them to optionally just pass in a `storageKey` or a `persistFunction`.

Reply:
@jaredhill4 where would it get persisted?